### DISCUSSION TITLE
404 Status for every page since version 13.4.13-canary.0

### DISCUSSION THREAD

**Discussion by fabio-nettis** [vercel/next.js#53667] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; root:xnu-8796.141.3~6/RELEASE_X86_64
    Binaries:
      Node: 20.5.0
      npm: 9.8.0
      Yarn: 3.6.1
      pnpm: N/A
    Relevant Packages:
      next: 13.4.13
      eslint-config-next: 13.4.13
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Describe the Bug

Since version 13.4.13-canary.0 an application previously working fine on version 13.4.12 returns a 404 status code for every page in the app folder and renders the 404 page instead. No console output is provided on the server or client side whatsoever. I have installed each 13.4.13-canary version and it has indeed stopped working from 13.4.13-canary.0 onwards.

### Expected Behavior

Defined pages obviously should still be accessible and not return a 404 error.

### Which browser are you using? (if relevant)

Chrome 114.0.5735.106 (Official Build) (x86_64)

### How are you deploying your application? (if relevant)

next dev

**Comment by fabio-nettis** [comment]
@balazsorban44 Why was this converted to a discussion? This issue was clearly introduced in 13.4.13-canary.0 and makes Next.js completely unusable. I'd be very happy about any feedback.

**Reply by balazsorban44** [reply]
There was no reproduction and also very limited information about the environment/code. We cannot investigate such issues. I also could not reproduce with a single hello world example, so it's likely a misconfiguration. For project-specific issues, we recommend using GitHub Discussions or checking out our [Discord server](https://discord.com/invite/bUG2bvbtHy) where community members can help look into your project more.

The Next.js issue tracker is for verifiable Next.js bugs only.

**Reply by fabio-nettis** [reply]
While the lack of a reproducible example may be a valid concern. I am not able to share the company repository nor to narrow it down to a specific code snipped per-se since it is a very large project and the issue not tracable to a specific change besides the upgrade to the new version. 

Also a misconfiguration is highly unlikely since as I stated above; when downgrading to version 13.4.12 again, the page works as expected and without any issues henceforth I went ahead and have tried each canary version leading up to the current stable release and the page stops working as soon as 13.4.13-canary.0.

Furthermore I went ahead an checked the diff for `13.4.12..13.4.13-canary.0`, from just looking at it I suspect that commit: [d8f4fa](https://github.com/vercel/next.js/commit/d8f4fa89466bf52c43abb7f89d1c139686775343) may have something to do with my issue, but again that is just a wild guess.

**Reply by balazsorban44** [reply]
It's hard to know what the issue can be without knowing anything about the project. It can be a wrong proxy, faulty Middleware, or something entirely different, based on the code. We have fairly thorough testing though, and this did not come up. If it's an actual bug, we would like to address it, but we have to know where to look.

If an issue does not have a proper reproduction, it might be converted to a discussion as stated in our Triaging documentation: https://github.com/vercel/next.js/blob/canary/contributing/repository/triaging.md

As well as in the bug report template itself:

![image](https://github.com/vercel/next.js/assets/18369201/ee66e47a-47db-4995-92e1-670559a68330)

Could you create a new issue with a minimal reproduction, please? :pray:

You can reduce your project by starting to remove pages/code that do not contribute to the issue to a point, where it's as minimal as possible, with as few dependencies as possible, so you can share it publicly. The idea is to eliminate everything irrelevant first to verify it's an actual Next.js bug. Ideally, a GitHub repository should be linked so that a maintainer can clone and easily verify without setting up third-party services, etc, unless irrelevant for the reproduction.

If you are hosting on Vercel, you might be able to open a Support Ticket as well.

**Comment by icyJoseph** [comment]
Got a reproduction demo? Or if its a private project, could you remove routes until it works again? It ought to be a route configuration issue, not that the route itself is wrong, but rather, not being handled, as expected, by the framework. Too early to say though.

**Reply by fabio-nettis** [reply]
Will eventually get to that today, problem is that no single page is accessible any longer.

**Comment by fabio-nettis** [comment, marked as answer]
So after about two hours of mindless debugging I finally came to a solution. It seems that removing the `i18n` setting inside the next config file as well as modifying our middleware  to thus no longer rely on `request.nextUrl.locale` has done the job.

```diff
module.exports = {
-  i18n: {
-    locales: ["default", "de", "fr", "it"],
-    defaultLocale: "default",
-  }
}
```

I have looked through the commits for the the diff `13.4.12..13.4.13-canary.0` again and sadly can not find any change that may have caused this error to occur or why the configuration suddenly stopped working the way it did. This may should be investigated further.

Opened issue: https://github.com/vercel/next.js/issues/53724

**Reply by icyJoseph** [reply]
Is that enough info to set up a create-next-app project that breaks reliably? If so, then you should open an issue linking to that project as a GitHub repository.

**Reply by fabio-nettis** [reply]
I will investigate that and try to break it again with a clean slate, if I still manage to break the app I will open a new issue with the gathered information.

**Reply by fabio-nettis** [reply]
@icyJoseph Got the [reproduction](https://github.com/fabio-nettis/next-i18n-error) repo ready. Opened issue: #53724

**Reply by balazsorban44** [reply]
Thank you for the investigation and the nice reproduction! The issue is now tracked, and looking into it! Feel free to comment on #53724

**Comment by fabio-nettis** [comment]
Until the issue I opened is resolved, I have written an [extensive guide](https://medium.com/@nettisfabio/the-ultimate-guide-to-internationalization-with-next-js-13-eb1be3d1a9e2) about Next.js 13 and internationalization, using the new app directory. Check it out! 🚀 The article already uses a workaround for the missing `i18n` key in `next.config.js`
